adapters/sinatra/lib/opentelemetry/adapters/sinatra/middlewares/tracer_middleware.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Adapters
+    module Sinatra
+      module Middlewares
+        class TracerMiddleware
+          def initialize(app)
+            @app = app
+          end
+
+          def call(env)
+            tracer.in_span(
+              env['PATH_INFO'],
+              attributes: { 'component' => 'http',
+                            'http.method' => env['REQUEST_METHOD'],
+                            'http.url' => env['PATH_INFO'] },
+              kind: :server,
+              with_parent_context: parent_context(env)
+            ) do |span|
+              app.call(env).tap { |resp| trace_response(span, env, resp) }
```

Comment:
dd-trace-rb has the ability to opt-in to collecting additional headers: https://github.com/DataDog/dd-trace-rb/blob/master/lib/ddtrace/contrib/faraday/middleware.rb#L45-L46.

Should we keep this functionality?

Reply:
Didn't see the opt-in on the lines you pointed at, but the [semantic conventions for HTTP clients](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-http.md#http-client) include the option of `peer.hostname` and `peer.port` or `peer.ip` and `peer.port`, so I think it's worth considering their inclusion here. That said, the semantic conventions favour `http.url`:

> One of the following sets of attributes is required (in order of usual preference unless for a particular web client/framework it is known that some other set is preferable for some reason; all strings must be non-empty):
>
> * http.url
> * http.scheme, http.host, http.target
> * http.scheme, peer.hostname, peer.port, http.target
> * http.scheme, peer.ip, peer.port, http.target

Reply:
I think you mean https://github.com/DataDog/dd-trace-rb/blob/ab135f65c4d761ddbac02d4d0f04e3c17382218f/lib/ddtrace/contrib/sinatra/tracer_middleware.rb#L30-L32

I'd lean toward "no" for the initial implementation, just to keep it simple and focussed on the current semantic specifications.

Reply:
lol... yes... that's the correct link